### PR TITLE
Add offsetX and offsetY to click event metadata

### LINKED ISSUE #493: Add offsetX and offsetY to click events
### Environment

* Phoenix LiveView version (mix deps): 0.4.1
### Actual behavior

At the moment a `click` sends an event with `pageX`, `pageY`, `screenX`, `screenY`, `x`, `y` coordinates. With these properties it's not possible to know the click coordinate relative to the clicked element.

### Expected behavior

 It would be also nice to have `offsetX` and `offsetY`, which are the click coordinates relative to the clicked element. 

This can be useful like when clicking on a big rectangular div where we want to render new elements.

offsetX_offsetY

I've already tried locally to add those two properties to the `meta` in [phoenix_live_view.js#L509](https://github.com/phoenixframework/phoenix_live_view/blob/master/assets/js/phoenix_live_view.js#L509) and works pretty well.

Before opening a PR, I wanted to know if it could be something useful or is it possible to get the same result in other ways without sending additional data in the event?

Thanks

Alvise
****

### PR DESCRIPTION
**Pull request by alvises**
resolves #493 

It adds `offsetX` and `offsetY` to click event metadata in `phoenix_live_view.js`, which are the click coordinates relative to the clicked element.

**A use case**: when clicking on a big rectangular div where we want to render new elements.

### CODE REVIEW

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -507,6 +507,9 @@ export class LiveSocket {
         pageY: e.pageY,
         screenX: e.screenX,
         screenY: e.screenY,
+        offsetX: e.offsetX,
+        offsetY: e.offsetY,
+
```
Comment (snewcomer) [inline assets/js/phoenix_live_view.js]: ✂️ extra space pleaze!

Comment (snewcomer) [reply assets/js/phoenix_live_view.js]: For future readers - these are CSS pixels from various points

`client` is from the current viewport
`page` is the encompassing browser window
`screen` is relative to the physical monitor (likely not very useful)
`offset` is relative to a parent container.

Comment (alvises) [reply assets/js/phoenix_live_view.js]: > ✂️ extra space pleaze!

done :-)

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Comment by alvises** [comment]
I've just published [an article](https://www.poeticoding.com/liveview-click-event-and-offsetx-offsety-coordinates/) on how to use these coordinates with an SVG element.

**Comment by robergperez** [comment]
Ironically, two months after alvies article, meta-data for phx-click became opt-in and must be defined in LiveSocket.[see here](https://hexdocs.pm/phoenix_live_view/bindings.html#click-events)

**Comment by codeanpeace** [comment]
Thanks @robergperez for your helpful comment!

For posterity's sake, the following restored `offsetX` & `offsetY` in the `handle_event/3` callback.

before:
`let liveSocket = new LiveSocket("/live", Socket, {params: {_csrf_token: csrfToken}})`

after:
```
let liveSocket = new LiveSocket("/live", Socket, {
  params: {_csrf_token: csrfToken},
  metadata: {
    click: (e, _el) => { return {offsetX: e.offsetX, offsetY: e.offsetY} }  
  }
})
```

**Pull request merged**